### PR TITLE
Fix several missing items in subscription dict

### PR DESCRIPTION
**Pull request by ryanchild**
In trying to figure out how cTiVo remembers which shows were downloaded
for a subscription, I came across MTSubscription.prevRecorded but
realized it was never getting saved.  Turns out this was because
kMTExportTextMetaData was never getting initialized so was always nil
for subscriptions.  Everything after it in the dictionary saved to
kMTSubscriptionList was getting lost, including exportSubtitles,
preferredTivo, HDOnly, SDOnly, and prevRecorded.

Simply initializing kMTExportTextMetaData in the user defaults fixes
this.

### CODE REVIEW

**Comment by ryanchild** [comment]
Reading through the code some more it looks like cTiVo only tries to download subscribed shows if the show was recorded between 12 hours before the last time cTiVo loaded a TiVo and now, so I guess this fix should prevent duplicate downloads if a show has been downloaded via subscription, deleted on the TiVo, and then re-recorded?  

This is kind of how I imagined myself using cTiVo's subscription's; set the TiVo to keep only 5 or so episodes and let cTiVo keep track of which episodes weren't already downloaded.

**Comment by mackworth** [comment]
Yeah, and I'm not clear how did this ever work for anyone (at least who didn't upgrade from iTiVo, as I did)?  

On your second point, the 12hrs thing is just an optimization. We want to avoid checking every show against every subscription every time we load up, so we remember when we last did it, and don't check any that were recorded before that. However, a show could have been recording while we last checked, and finished recording afterwards. Thus we could not have downloaded it that last time, so we need to check it now.  Figuring no show is longer than 12 hours, leads to the code you see.

On the other hand, you're correct that if prevShows was getting dropped, then from one execution run to the next, it would forget which shows were recorded and re-record any it sees. Fortunately, the optimization above was preventing it from doing ALL the old shows, and if the successful download was still in the queue, it would not recreate it.

In other words, thanks for a great catch.  

"Nil"s silently failing is certainly an interesting choice. Can't wait to move on to Swift.
